src/dmd/backend/blockopt.d

```diff
@@ -652,196 +711,196 @@ void blockopt(int iter)
+                    list_subtract(&(bsucc.Bpred),b3);
+                    debug if (debugc) printf("brcombine(): if e1 goto e2 else goto e3 => ret e1?e2:e3\n");
                     anychanges++;
                 }
             }
             else if (bc == BCgoto && PARSER)
             {
-                block *b2 = b->nthSucc(0);
-                if (!list_next(b2->Bpred) && b2->BC != BCasm    // if b is only parent
+                block *b2 = b.nthSucc(0);
+                if (!list_next(b2.Bpred) && b2.BC != BCasm    // if b is only parent
                     && b2 != startblock
-                    && b2->BC != BCtry
-                    && b2->BC != BC_try
-                    && b->Btry == b2->Btry
+                    && b2.BC != BCtry
+                    && b2.BC != BC_try
+                    && b.Btry == b2.Btry
                    )
                 {   list_t bl;
 
-                    if (b2->Belem)
+                    if (b2.Belem)
                     {
                         if (PARSER)
                         {
-                            block_appendexp(b,b2->Belem);
+                            block_appendexp(b,b2.Belem);
                         }
-                        else if (b->Belem)
-                            b->Belem = el_bin(OPcomma,b2->Belem->Ety,b->Belem,b2->Belem);
+                        else if (b.Belem)
+                            b.Belem = el_bin(OPcomma,b2.Belem.Ety,b.Belem,b2.Belem);
                         else
-                            b->Belem = b2->Belem;
-                        b2->Belem = NULL;
+                            b.Belem = b2.Belem;
+                        b2.Belem = null;
                     }
-                    list_subtract(&b->Bsucc,b2);
-                    list_subtract(&b2->Bpred,b);
+                    list_subtract(&b.Bsucc,b2);
+                    list_subtract(&b2.Bpred,b);
 
                     /* change predecessor of successors of b2 from b2 to b */
-                    for (bl = b2->Bsucc; bl; bl = list_next(bl))
+                    for (bl = b2.Bsucc; bl; bl = list_next(bl))
                     {   list_t bp;
 
-                        for (bp = list_block(bl)->Bpred; bp; bp = list_next(bp))
+                        for (bp = list_block(bl).Bpred; bp; bp = list_next(bp))
                         {
                             if (list_block(bp) == b2)
-                                list_ptr(bp) = (void *)b;
+                                bp.ptr = cast(void *)b;
                         }
                     }
 
-                    b->BC = b2->BC;
-                    b->BS = b2->BS;
-                    b->Bsucc = b2->Bsucc;
-                    b2->Bsucc = NULL;
-                    b2->BC = BCret;             /* a harmless one       */
-                    cmes3("brcombine(): %p goto %p eliminated\n",b,b2);
+                    b.Bswitch = b2.Bswitch;
+                    b.Bswitch = b2.Bswitch;
```

Comment:
I'm not sure whether this is the correct translation.

Reply:
This is copying too little, it should be somethink llike
```
                    b.BC = b2.BC;
                    b.cover_BS[] = b2.cover_BS[];
```
with
```
        // add member at least as large as the other elements of this union
        void[jcatchvar.sizeof + Bscope_index.sizeof + Blast_index.sizeof] cover_BS;
```
added to the respective union in `block`.